### DISCUSSION TITLE
@apply is not supported within nested at-rules like @nest

### DISCUSSION THREAD

**Discussion by northern-mindset** [tailwindlabs/tailwindcss#7683] [Help]
Upgrading v2 to v3

#### Error message

```
@parcel/transformer-postcss: /path/to/my.css:190:5: @apply is not supported     
within nested at-rules like @nest. You can fix this by un-nesting @nest.
```

#### `package.json`

```
    "parcel": "2.2.1",
    "postcss": "8.4.6",
    "postcss-mixins": "9.0.1",
    "postcss-nesting": "10.1.2",
    "tailwindcss": "3.0.23"
```

#### `.postcssrc`
```json
{
  "plugins": {
    "postcss-mixins": true,
    "tailwindcss/nesting": "postcss-nesting",
    "tailwindcss": true,
    "autoprefixer": true
  }
}
```

#### CSS file
```css
  .ais-HierarchicalMenu-link {
    @nest .ais-HierarchicalMenu-item--parent > &:after {
      @apply inline-block;
    }

    @nest .ais-HierarchicalMenu-item--selected > &:after {
      @apply rotate-90;
    }
  }
```

#### Observations

- Other nested rules (except explicit `@nest`) are compiled properly.
- Removing `@nest` keyword fixes the issue.

Based on this, I guess that `tailwindcss/nesting` may be not catching the `postcss-nesting` plugin and still uses `postcss-nested`, which implements another nesting specification and thus does not care for `@nest`. But why? I even tried the following PostCSS configuration file to pass the plugin itself instead of its name, but to the same effect:

```js
module.exports = {
  plugins: {
    'postcss-mixins': true,
    'tailwindcss/nesting': require('postcss-nesting'),
    tailwindcss: true,
    autoprefixer: true,
  },
};
```

There may be something wrong with the [documentation](https://tailwindcss.com/docs/using-with-preprocessors#:~:text=Then%20pass%20the%20plugin%20itself%20as%20an%20argument%20to%20tailwindcss/nesting%20in%20your%20PostCSS%20configuration), which says:

> Then pass the plugin itself as an argument to tailwindcss/nesting in your PostCSS configuration

but in the example they pass the **plugin name** as an argument.

```
'tailwindcss/nesting': 'postcss-nesting'
```

Anyhow, neither approach seem to work in my case.

**Comment by greg-gorlen-andela** [comment]
Related Stack Overflow thread: [@apply is not supported within nested at-rules like @mixin](https://stackoverflow.com/questions/77819999/apply-is-not-supported-within-nested-at-rules-like-mixin)
